### PR TITLE
Add requirements listing in new UI

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Add requirements listing in new UI
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | -

### CODE REVIEW

File: classes/Commands/CheckRequirementsCommand.php [hunk 1]
```diff
@@ -114,47 +116,25 @@ protected function execute(InputInterface $input, OutputInterface $output): ?int
...
-                   $this->writeError($this->upgradeSelfCheck->getRequirementWording($error));
-                   foreach ($notWritingDirectories as $notWritingDirectory) {
-                       $this->output->writeln('    ' . $notWritingDirectory);
-                   }
-                   break;
-               default:
-                   $this->writeError($this->upgradeSelfCheck->getRequirementWording($error));
-           }
+            $wording = $this->upgradeSelfCheck->getRequirementWording($error);
+            $this->writeError($wording['value']);
```
Comment (ga-devfront) [inline classes/Commands/CheckRequirementsCommand.php]: what do you think about `message` instead of `value` ?

File: classes/UpgradeSelfCheck.php [hunk 2]
```diff
@@ -194,101 +195,156 @@ public function getWarnings(): array
...
+                } else {
+                    $params = [
+                        '[1]' => '',
+                        '[/1]' => '',
+                    ];
+                }
+
+                return [
+                    'value' => $this->translator->trans('Maintenance mode needs to be enabled. Enable maintenance mode and add your maintenance IP in [1]Shop parameters > General > Maintenance[/1].', $params),
+                    'list' => [],
```
Comment (ga-devfront) [inline classes/UpgradeSelfCheck.php]: why do you add empty list ? I think we can just add list if needed no ?

**Review by ga-devfront — DISMISSED** [review]
I think all since good for me got just few comments

File: classes/UpgradeSelfCheck.php (hunk 2 above)
Comment (M0rgan01) [reply classes/UpgradeSelfCheck.php]: This saves me from having to check if there is a list in the command or in the template.

Comment (M0rgan01) [reply classes/UpgradeSelfCheck.php]: Afterwards, it could be cleaner. I'm going to make this optional

**Review by ga-devfront — DISMISSED** [review]

**Review by ga-devfront — APPROVED** [review]

**Review by ga-devfront — DISMISSED** [review]

**Review by ga-devfront — DISMISSED** [review]

**Review by AureRita — APPROVED** [review]
Hi @M0rgan01 

Thank you for your PR, I tested it and it seems to works as you can see : 

image


Tested from : 
1.7.8.10 to 8.2 
8.0.5 to 8.2 
8.2 to 9.0.0
8.0.5 to 9.0.0

We also see that we doesn't have scroll bar when we have to much error as you can see : 

image


Because this one'll be corrected in other PR, this PR seems to works as expected, It's QA :heavy_check_mark: 

Thank you

**Pull request merged**